node/src/main/scala/coop/rchain/node/api/WebApi.scala

```diff
@@ -119,7 +120,10 @@ object WebApi {
       blockApi.isFinalized(hash).flatMap(_.liftToBlockApiErr)
 
     def getTransaction(hash: String): F[TransactionResponse] =
-      cacheTransactionAPI.getTransaction(hash)
+      OptionT
+        .whenF(hash.nonEmpty)(cacheTransactionAPI.getTransaction(hash))
+        .value
+        .flatMap(_.liftTo(new Exception("Block hash cannot be empty.")))
```

Comment:
Nice solution!

Here we want to throw `BlockApiException` which will not be logged because it's related to invalid user input.